File: csharp/bird-watcher/BirdWatcher.cs
using System;
using System.Linq;

class BirdCount
{
    private int[] birdsPerDay;

    public BirdCount(int[] birdsPerDay)
    {
        this.birdsPerDay = birdsPerDay;
    }

    public static int[] LastWeek() => new[] { 0, 2, 5, 3, 7, 8, 4 };

    public int Today() => birdsPerDay.Last();

    public void IncrementTodaysCount()
    {
        ++birdsPerDay[birdsPerDay.Length - 1];
    }

    public bool HasDayWithoutBirds() => birdsPerDay.Any(x => x == 0);

    public int CountForFirstDays(int numberOfDays) =>
        Enumerable.Range(0, numberOfDays).Select(i => birdsPerDay[i]).Sum();

    public int BusyDays() => birdsPerDay.Where(x => x >= 5).Count();
}

File: java/bird-watcher/src/main/java/BirdWatcher.java
import java.util.Arrays;

class BirdWatcher {
  private final int[] birdsPerDay;

  public BirdWatcher(int[] birdsPerDay) {
    this.birdsPerDay = birdsPerDay;
  }

  public int[] getLastWeek() {
    return new int[] {0, 2, 5, 3, 7, 8, 4};
  }

  public int getToday() {
    return birdsPerDay[birdsPerDay.length - 1];
  }

  public void incrementTodaysCount() {
    ++birdsPerDay[birdsPerDay.length - 1];
  }

  public boolean hasDayWithoutBirds() {
    return Arrays.stream(birdsPerDay).anyMatch(x -> x == 0);
  }

  public int getCountForFirstDays(int numberOfDays) {
    return Arrays.stream(birdsPerDay).limit(numberOfDays).sum();
  }

  public int getBusyDays() {
    return (int) Arrays.stream(birdsPerDay).filter(x -> x >= 5).count();
  }
}
